daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -44,37 +51,23 @@ private[lf] object Pretty {
         prettyScenarioError(serr)
     })
 
-  def prettyError(err: SError): Doc = {
-    val ptx = PartialTransaction.initial(
-      (_ => TransactionVersion.minVersion),
-      ContractKeyUniquenessMode.On,
-      submissionTime = Time.Timestamp.MinValue,
-      initialSeeds = InitialSeeding.NoSeed,
-    )
-    prettyError(err, ptx)
-  }
-
   def prettyParty(p: Party): Doc =
     char('\'') + text(p) + char('\'')
 
-  private def prettyDamlException(ex: SErrorDamlException, ptx: PartialTransaction): Doc =
-    ex match {
-      case DamlEFailedAuthorization(nid, fa) =>
+  def prettyDamlException(error: interpretation.Error, ptx: PartialTransaction = dummyTx): Doc = {
```

Comment:
side note: I’m not convinced the partial transaction here provides any real value. Afaict the only thing we use it for is to produce slightly different error if a contract is inactive due to a recursive exercise. We don’t do that in the scenario service however and I’ve never heard a request for it so I have a hard time believing that it is really useful here.